blocks.js

```diff
@@ -7553,13 +7554,38 @@ InputSlotMorph.prototype.dropDownMenu = function (enableKeyboard) {
         if (Object.prototype.hasOwnProperty.call(choices, key)) {
             if (key[0] === '~') {
                 menu.addLine();
-            // } else if (key.indexOf('§_def') === 0) {
-            //     menu.addItem(choices[key].blockInstance(), choices[key]);
+        // } else if (key.indexOf('§_def') === 0) { 
+        // menu.addItem(choices[key].blockInstance(), choices[key]);
+            } else if (key.charCodeAt(key.length-1) == 0x25ba) {
+                // Submenu
+                menu.addItem(key, choices[key]);
+                menustack.push(menu);
+                menu = new MenuMorph(
+                                     this.setContents,
+                                     null,
+                                     this,
+                                     this.fontSize
+                                 );
+            } else if (key.charCodeAt(0) == 0x25c4) {
+                var oldmenu=menustack[menustack.length-1],
+                    items=oldmenu.items;
+                items[items.length-1][6] = menu;
```

Comment:
I don't think you can easily do something about this, but accessing only item 6 of an array is troublesome. Loads of room for error, and it's not very communicative.